pkg/network/ebpf/c/prebuilt/http.c

```diff
@@ -239,6 +240,71 @@ int uprobe__SSL_shutdown(struct pt_regs* ctx) {
     return 0;
 }
 
+SEC("kprobe/do_sys_open")
+int kprobe__do_sys_open(struct pt_regs* ctx) {
+    char *path_argument = (char *)PT_REGS_PARM2(ctx);
+    lib_path_t path = {0};
+    bpf_probe_read(path.buf, sizeof(path.buf), path_argument);
+
+    // Find the null character and clean up the garbage following it
+#pragma unroll
+    for (int i = 0; i < LIB_PATH_MAX_SIZE; i++) {
+        if (path.len) {
+            path.buf[i] = 0;
+        } else if (path.buf[i] == 0) {
+            path.len = i;
+        }
+    }
+
+    // Bail out if the path size is larger than our buffer
+    if (!path.len) {
+        return 0;
+    }
+
+    u64 pid_tgid = bpf_get_current_pid_tgid();
+    bpf_map_update_elem(&open_at_args, &pid_tgid, &path, BPF_ANY);
+    return 0;
+}
+
+SEC("kretprobe/do_sys_open")
+int kretprobe__do_sys_open(struct pt_regs* ctx) {
+    u64 pid_tgid = bpf_get_current_pid_tgid();
+
+    // If file couldn't be opened, bail out
+    if (!(long)PT_REGS_RC(ctx)) {
+        goto cleanup;
+    }
+
+    lib_path_t *path = bpf_map_lookup_elem(&open_at_args, &pid_tgid);
+    if (path == NULL) {
+        return 0;
+    }
+
+    // Detect whether the file being opened is a shared library
+    int is_shared_library = 0;
```

Comment:
You should be able to use a `bool` type here. https://github.com/torvalds/linux/blob/fc0c0548c1a2e676d3a928aaed70f2d4d254e395/include/linux/types.h#L30